### PR TITLE
AC_PosControl: init with target attitude not actual

### PR DESCRIPTION
**Pull request by IamPete1**
This implements two todo's from the position controller, this should remove the issue seen on TVBS takeoff when there is a very large pitch angle error.

https://github.com/ArduPilot/ardupilot/pull/16134

### CODE REVIEW

**Comment by IamPete1** [comment]
just double checked the S Curves PR, there is no conflict with this

**Comment by Hwurzburg** [comment]
Seems to work as desired in SITL...thanks!

**Review by kd0aij — APPROVED** [review]
NIce, looks like this is the right way to fix the TVBS issue and tests OK in RF8/sitl with the SkyCat.

closes #16134

**Comment by lthall** [comment]
How come this was merged without my approval?

This breaks multirotor!!

File: libraries/AC_AttitudeControl/AC_PosControl.cpp
Comment (lthall) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp:1139]: We currently base the acceleration vector on the real heading, not the current heading. This prevents the aircraft from toilet bowling if we have a yaw error. This happens often for multirotors and especially when we lose a motor.

I have been looking at a navigation input function that defines the thrust vector as the primary command leaving yaw as unconditional lower priority axis. However the change didn't fit well with our current code structure so I felt it would be better left for now. So we are stuck with an annoying mix of euler angle order side effects and a mix of euler and measured angles to get this right.

**Review by lthall — CHANGES_REQUESTED** [review]
This is a really nice enhancement.

I can't comment on mechanics and programming rules around the "friend class AC_PosControl;" lines. I have not seen this before so I would like Randy to be happy with the way it has been done.

We just need to deal with the target vs measured yaw issue and I think it is good. (and the fact we need to deal with that doesn't feel right to me either but I have not worked out exactly what the right solution is yet)

File: libraries/AC_AttitudeControl/AC_PosControl.cpp
Comment (rmackay9) [reply libraries/AC_AttitudeControl/AC_PosControl.cpp:1139]: I think we can pretty easily avoid the "friend class" addition by reusing get_att_target_euler_cd() in this "lean_angles_to_accel" method.  If the extra conversion to and from degrees is a problem then let's instead add another new method called "get_att_target_euler_rad()" that simply returns "_attitude_target_euler_angle".

**Comment by rmackay9** [comment]
Great!  I've put a suggestion above but once that's fixed hopefully we can merge this in.

@IamPete1 I'm sure you've tested this on a multicopter in SITL?  I'm sure you must have 'cuz it can't have been trivial to get the calcs right.

File: libraries/AC_AttitudeControl/AC_PosControl.cpp
Comment (IamPete1) [reply libraries/AC_AttitudeControl/AC_PosControl.cpp:1139]: OK, I will remove the friend class and add the new method.

**Comment by IamPete1** [comment]
@lthall I have changed to use true yaw, this seems to work OK in real-flight, although I was having a hard time tying to reproduce a worst case test with a large yaw error.

**Review by lthall — APPROVED** [review]
Looks good.

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h
Comment (peterbarker) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.h:183]: This could be returning a reference.

File: libraries/AC_AttitudeControl/AC_PosControl.cpp
Comment (peterbarker) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp:828]:
```suggestion
    const Vector3f &att_target_euler_cd = _attitude_control.get_att_target_euler_cd();
```

File: libraries/AC_AttitudeControl/AC_PosControl.cpp
Comment (peterbarker) [inline libraries/AC_AttitudeControl/AC_PosControl.cpp:1132]:
```suggestion
    const Vector3f &att_target_euler = _attitude_control.get_att_target_euler_rad();
```

Comment (IamPete1) [reply libraries/AC_AttitudeControl/AC_PosControl.cpp]: done

File: libraries/AC_AttitudeControl/AC_PosControl.cpp
Comment (IamPete1) [reply libraries/AC_AttitudeControl/AC_PosControl.cpp]: done

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h
Comment (IamPete1) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.h]: done

**Comment by rmackay9** [comment]
Merged, thanks!

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h
Comment (rmackay9) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.h:183]: There's a little whitespace issue with the "&" but we can get it at some future date.

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**
